realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProcessor.java

```diff
@@ -122,55 +123,72 @@
         "io.realm.annotations.Required"
 })
 public class RealmProcessor extends AbstractProcessor {
-
     // Don't consume annotations. This allows 3rd party annotation processors to run.
     private static final boolean CONSUME_ANNOTATIONS = false;
 
-    Set<ClassMetaData> classesToValidate = new HashSet<ClassMetaData>();
+
+    // List of all fields maintained by Realm (RealmResults)
+    private final Set<ClassMetaData> classesToValidate = new HashSet<ClassMetaData>();
+    // List of backlinks
+    private final Set<Backlink> backlinksToValidate = new HashSet<Backlink>();
+
     private boolean hasProcessedModules = false;
+    private int round;
 
     @Override public SourceVersion getSupportedSourceVersion() {
         return SourceVersion.latestSupported();
     }
 
     @Override
     public boolean process(Set<? extends TypeElement> annotations, RoundEnvironment roundEnv) {
-        // Don't run this processor in subsequent runs. We created everything in the first one.
-        if (hasProcessedModules) {
-            return CONSUME_ANNOTATIONS;
+        round++;
+
+        if (round == 0) {
+            RealmVersionChecker updateChecker = RealmVersionChecker.getInstance(processingEnv);
+            updateChecker.executeRealmVersionUpdate();
         }
-        RealmVersionChecker updateChecker = RealmVersionChecker.getInstance(processingEnv);
-        updateChecker.executeRealmVersionUpdate();
 
-        Utils.initialize(processingEnv);
+        if (roundEnv.errorRaised()) { return true; }
 
-        Set<String> packages = new TreeSet<String>();
+        if (!hasProcessedModules) {
+            Utils.initialize(processingEnv);
 
-        // Create all proxy classes
+            if (!processAnnotations(roundEnv)) { return true; }
+
+            hasProcessedModules = true;
+            if (!processModules(roundEnv)) { return true; }
+        }
+
+        if (roundEnv.processingOver()) {
+            if (!validateBacklinks()) { return true; }
+        }
+
+        return CONSUME_ANNOTATIONS;
+    }
+
+    // Create all proxy classes
+    private boolean processAnnotations(RoundEnvironment roundEnv) {
         for (Element classElement : roundEnv.getElementsAnnotatedWith(RealmClass.class)) {
 
             // The class must either extend RealmObject or implement RealmModel
             if (!Utils.isImplementingMarkerInterface(classElement)) {
-                Utils.error("A RealmClass annotated object must implement RealmModel or derive from RealmObject", classElement);
+                Utils.error("A RealmClass annotated object must implement RealmModel or derive from RealmObject.", classElement);
             }
```

Comment:
Are you sure?  This is exactly how the code looks at the head of master.  I assumed it was some attempt to give as many error messages as possible, before aborting...